src/types.cpp

```diff
@@ -661,6 +661,8 @@ namespace Exiv2 {
     {
 #if defined(_MSC_VER) && _MSC_VER < 1800
         #define isinf(x) (!_finite(x))
+#elif __APPLE__
```

Comment:
Are you sure that `__APPLE__` is defined to a value that evaluates to true? If it is merely defined, then this check could fail.

Reply:
I think so. I have always used it like that in other projects successfully. Let's merge it like this and we can always come back to it if we see an issue like the one you mentioned.